create a new policies page (#1893)
semgrep/semgrep-docs: docs/semgrep-supply-chain/policies.md

---
slug: policies
title: Manage policies
hide_title: true
description: Use policies to define the conditions in which developers are notified of a finding or potentially blocked from merging their PR or MR.
tags:
  - Semgrep Supply Chain
  - Semgrep AppSec Platform
---

# Manage policies (beta)

Use Supply Chain policies to define the conditions in which developers are **notified** of a finding through a comment, or potentially **blocked** from merging their pull or merge request (PR or MR). This feature helps you manage noise and ensures that developers are only notified or potentially blocked based on the conditions you set.

This feature enables you to configure the following:

- **Scope**: These are the projects (repositories) that are affected by the policy.

> - **Conditions**: The conditions under which **actions** are performed. These conditions are typically related to the value of a finding's severity or reachability.

Reviewer: I liked that you included examples of what could be an example of a condition, though I wonder if we could frame it as something like examples of finding attributes that can be defined as a condition. So something like the following maybe?
```suggestion
- **Conditions**: The conditions under which **actions** are performed. These conditions are typically attributes of a finding such as severity or reachability. 
```

- **Actions**: Actions that are performed on the defined scope when conditions are met.

You can create as many policies as you need.

## Feature maturity

The Supply Chain policies feature is in **private beta**. To apply to this beta, reach out to [<i class="fa-regular fa-envelope"></i> support@semgrep.com](mailto:support@semgrep.com).

## Prerequisites

This feature requires the `semgrep:latest` Docker image or at least version 1.101.0 of the Semgrep CLI tool.

## View your policies

Only **admins** can view, create, edit, or delete policies.

1. Sign in to [<i class="fas fa-external-link fa-xs"></i> Semgrep AppSec Platform](https://semgrep.dev/login).
1. From the navigation menu, click **Rules** to expand the drop-down box, then click **Policies**.
1. Click **Supply Chain**. This takes you to the Supply Chain policies tab.

## Create a policy

1. From the Supply Chain policies tab, Click **Create policy**.
1. Provide a **Policy name**.
1. Define the scope of the policy:
  1. Click the drop-down box to select between **All Projects**, **Project**, or **Project tag**. 

>   1. For **Project** or **Project tag** values, a second drop-down box appears. Choose the **projects** or **project tags** to finish defining the scope. You can select more than one.

Reviewer: I'm checking out the live preview and it seemed like these may have not been indented properly (though I'm going off of the live preview you linked in your slack message so it might be old). Currently seeing this in the live preview:
<img width="1034" alt="image" src="https://github.com/user-attachments/assets/252d61a1-5371-4d0b-b0f6-7d18d5bf6dc5" />

Author: Yup, looks like it requires a 4-space indent. Nice catch, thank you.

>   1. For **Project** or **Project tag** values, a second drop-down box appears. Choose the **projects** or **project tags** to finish defining the scope. You can select more than one.

Reviewer: super nit: "You can select more than one" reads to me that you can select more than one type of value at once, which isn't quite true. I know that you're referring to how we can select more than one project / one project tag at a time, but I wonder if there's an alternate way of phrasing that?

Author: On second look, "select more than one" is redundant since "projects" and "project tags" are already plural. I have just removed it. The previous step already states that the user must select only one (all projects OR project tags OR project) so that should help as well.

> 1. Define the conditions of the policy by selecting either **Reachable** or **Critical or high severity, reachable, upgrades available**. The latter value is a broader set of conditions and typically results in more findings shown to developers.

Reviewer: ```suggestion
1. Define the conditions of the policy by selecting either **Reachable** or **Critical or high severity, reachable, upgrades available**. The former value is a broader set of conditions and typically results in more findings shown to developers.
```

I think the latter actually defines a more narrow scope of conditions since the finding must be reachable in addition to the rest of the defined conditions.

1. Define the actions of the policy. You can choose to **Leave a comment** or **Block and leave a comment**.
1. Click **Save**. This brings you back to the Supply Chain policies tab.

## Enable a policy

> Creating a policy does **not** enable it by default. To enable it, click the **toggle** on the policy's entry in the Supply Chain policies tab.

Author: :ACK: thanks to Kevin and Lisa for considering my suggestion and you for keeping me updated. :)

> Creating a policy does **not** enable it by default. To enable it, click the **toggle** on the policy's entry in the Supply Chain policies tab.

Author: Resolved in latest commit.
